### ISSUE TITLE
`sde test` fails to build model when a spec file is provided

### ISSUE THREAD

**Issue by chrispcampbell** [climateinteractive/SDEverywhere#35]
Some of the `sde` CLI files, such as `sde-build.js`, have an async handler function, which works fine when that command is called directly (e.g. `sde build --spec spec.json`) but other commands are implemented by chaining and can fail when a spec file is passed on the command line.  The spec file is read in an async function, but some of the chaining cases do not use async/await.

For example:
* `sde test` calls `run` (which is currently not marked `async`)
* `run` then calls `build` (also not marked `async`)
* `build` calls `generate` (which is marked `async`)

Long story short, we need to add `async` and `await` in a couple more places to ensure that the spec file is read fully before proceeding with each step.

**Comment by chrispcampbell** [comment]
Fixed on `develop` in afdbb77.
